### ISSUE TITLE
Uses deprecated python-webkit

### ISSUE THREAD

**Issue by bubulle** [pytrainer/pytrainer#40]
The Debian BTS will probably explain this much better than me : 

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=749256

Package: pytrainer
Version: 1.10.1-2
Severity: important
User: pkg-webkit-maintainers@lists.alioth.debian.org
Usertags: oldlibs python-webkit webkit1

Hi,

pytrainer depends on python-webkit, which is deprecated and going to
be removed from the archive. pytrainer should switch to using the
GObject Introspection bindings for webkit, ideally gir1.2-webkit2-3.0
for Webkit2 and gtk+3 as upstream has already dropped WebKit1 (so
gir1.2-webkit-1.0 and gir1.2-webkit-3.0 are likely to be dropped from
Debian in the near future). This would mean switching to GObject
Introspection for other bindings as well, e.g. GLib and GTK+, so
making the switch to GTK+ 3 at the same time would make a lot of
sense.

For more information on GObject Introspection see [1] and [2].

If you have any question don't hesitate to ask.

Emilio

[1] https://wiki.gnome.org/action/show/Projects/GObjectIntrospection
[2] https://wiki.gnome.org/action/show/Projects/PyGObject

**Comment by bubulle** [comment]
Just got this reminder from Debian BTS:
pytrainer 1.10.2~git20160107-1 is marked for autoremoval from testing on 2016-03-17

It (build-)depends on packages with these RC bugs:
790218: pywebkitgtk: depends on libwebkitgtk-1.0-0 which is deprecated

In short, it means that pytrainer won't be part of the next Debian release as it relies on a component that is deprecated and now scheduled for removal from the distribution.

**Comment by viiru-** [comment]
I think the solution to this is as follows:

* First migrate the rest of the code from glade to gtk.Builder
* Then migrate from pygtk to gobject-introspection
* And finally replace the embedded browser with osmgpsmap

A lot of work, any help would be appreciated.

**Comment by MeisterP** [comment]
Gnome uses libchamplain (https://wiki.gnome.org/Projects/libchamplain). So you might consider using that instead of osmgpsmap.

**Comment by avtobiff** [comment]
Isn't this fixed by #120?

Although the migration to either osmgpsmap or libchamplain might be an option still.

**Comment by viiru-** [comment]
@avtobiff Yes, the deprecation issue was fixed by #120. However I still think the current map component isn't the right way to do this (embedding a web browser that is used to send the route info to Google Maps for rendering does not seem right at all).

The current consensus is to delay properly fixing this (rewriting the map functionality) until after 2.0, and I have removed this from the milestone to reflect that.
